server/src-lib/Hasura/RQL/DDL/Schema.hs

```diff
@@ -56,24 +59,35 @@ import           Hasura.Server.Utils            (matchRegex)
 data RunSQL
   = RunSQL
   { rSql                      :: Text
-  , rCascade                  :: !(Maybe Bool)
+  , rCascade                  :: !Bool
   , rCheckMetadataConsistency :: !(Maybe Bool)
+  , rReadOnly                 :: !Bool
```

Comment:
Since we have the custom `FromJSON` instance now, anyway, why not make the type of this field `Q.TxAccess` instead of `Bool`? 😄 Or if you’d rather not need to write the `ToJSON` instance by hand, you could add a `newtype IsReadOnly = IsReadOnly (Q.TxAccess)` with custom `ToJSON`/`FromJSON` instances, like we already have with the `SystemDefined` type.

Reply:
I think this comment is still relevant?

Reply:
Oops. Missed it, yes.

Reply:
I tried to make this change but it throws that `Lift` is not defined for `Q.TxAccess`, here is the patch for your reference: https://github.com/tirumaraiselvan/graphql-engine/pull/74/files

Reply:
I’ve pushed https://github.com/hasura/pg-client-hs/commit/bd21e66d8197af381a6c0b493e22d611ed1fa386, which derives a `Lift` instance for `TxAccess`.